be/src/storage/lake/lake_persistent_index.cpp

```diff
@@ -14,40 +14,119 @@
 
 #include "storage/lake/lake_persistent_index.h"
 
+#include "fs/fs_util.h"
+#include "storage/lake/filenames.h"
 #include "storage/lake/persistent_index_memtable.h"
+#include "storage/lake/tablet_manager.h"
+#include "storage/sstable/persistent_index_sstable.h"
 
 namespace starrocks::lake {
 
-LakePersistentIndex::LakePersistentIndex(std::string path) : PersistentIndex(std::move(path)) {
-    _memtable = std::make_unique<PersistentIndexMemtable>();
-}
+LakePersistentIndex::LakePersistentIndex(TabletManager* tablet_mgr, int64_t tablet_id)
+        : PersistentIndex(""),
+          _memtable(std::make_unique<PersistentIndexMemtable>()),
+          _tablet_mgr(tablet_mgr),
+          _tablet_id(tablet_id) {}
 
 LakePersistentIndex::~LakePersistentIndex() {
     _memtable->clear();
+    _sstables.clear();
+}
+
+bool LakePersistentIndex::is_memtable_full() {
+    const auto memtable_mem_size = _memtable->memory_usage();
+    return memtable_mem_size >= config::lake_persistent_index_memtable_size;
+}
+
+void LakePersistentIndex::flush_to_immutable_memtable() {
+    _immutable_memtable = std::move(_memtable);
+    _memtable = std::make_unique<PersistentIndexMemtable>();
```

Comment:
```
_immutable_memtable = std::make_unique<PersistentIndexMemtable>();
_memtable.swap(_immutable_memtable);
```

Reply:
whether need to report error if the `_immutable_memtable` is not nullptr for any reason?

Reply:
if minor_compaction is always expected before flush_to_immutable_memtable, will it be better to move the minor_compaction inside flush_to_immutable_memtable, so the two ops can be always together and the misuse of flush_to_immutable_memtable can be avoided.